### PR TITLE
Fix integer overflow calculating length of very large playlists

### PR DESCRIPTION
**Pull request by jcorporation**

### CODE REVIEW

**Comment by MaxKellermann** [comment]
This loses type safety by downgrading from std::chrono to raw integers, but what do these raw integers mean? You make the core more obscure and less safe.

**Comment by jcorporation** [comment]
Ok, migrated it to std::chrono::duration api.

File: src/playlist/Length.cxx [hunk 1]
```diff
@@ -36,15 +36,16 @@ playlist_provider_length(Response &r,
...
 	while ((song = e.NextSong()) != nullptr) {
 		if (playlist_check_translate_song(*song, base_uri,
 						  loader))
 			playtime += get_duration(*song);
 		i++;
 	}
 	r.Fmt(FMT_STRING("songs: {}\n"), i);
-	r.Fmt(FMT_STRING("playtime: {}\n"), playtime.RoundS());
+	const auto seconds = std::chrono::duration_cast<std::chrono::seconds>(playtime);
+	r.Fmt(FMT_STRING("playtime: {}\n"), seconds.count());
```
Comment (MaxKellermann) [inline src/playlist/Length.cxx:48]: This removes the rounding which was very explicit previously

File: src/playlist/Length.cxx [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 static SignedSongTime get_duration(const DetachedSong &song) {
 	const auto duration = song.GetDuration();
-	return duration.IsNegative() ? (SignedSongTime)0 : song.GetDuration();
+	return duration.IsNegative() ? (SignedSongTime)0 : duration;
```
Comment (MaxKellermann) [inline src/playlist/Length.cxx:24]: ?

Comment (jcorporation) [reply src/playlist/Length.cxx:24]: As I get the song duration in the local variable there is no need to call the GetDuration function a second time.

File: src/playlist/Length.cxx (hunk 1 above)
Comment (jcorporation) [reply src/playlist/Length.cxx]: Replaced with std::chrono::round

File: src/playlist/Length.cxx (hunk 2 above)
Comment (MaxKellermann) [reply src/playlist/Length.cxx:24]: True, but it doesn't fix an integer overflow

**Pull request merged**